kubernetes-manifests/hello.deployment.yaml

```diff
@@ -13,7 +13,7 @@ spec:
   containers:
   - name: myapp-container
     image: busybox:1.28
-    command: ['sh', '-c', 'echo $(cat /tmp/secret) && sleep 3600']
+    command: ['sh', '-c', 'echo $(cat /tmp/CATS_N_DOGS) && sleep 3600']
```

Comment:
Shouldn't this be MOUNT_NAME?

Reply:
Not sure I follow. It is the value of MOUNT_NAME defined on the init container. The init container will mount the secret to `/tmp/<MOUNT_NAME>`. For this demo the MOUNT_NAME is set to `CATS_N_DOGS`. Therefore if we want the `myapp-container` to show our secret it should echo `/tmp/CATS_N_DOGS`.

Reply:
If you want to replace an environment variable in a shell with its value, you need preface it with a $, e.g. $VAR_NAME. If you want to replace the ENV with its value, you use the ENV name, not its value.

Reply:
Right. The environment variable is available in the init container not the app container in this example. We could certainly add the environment variable to the container as well. I think this example is more illustrative of how it works.

Reply:
Maybe I misunderstand the flow:

- Init container runs with configuration items.
- Init container reads secret from AWS then writes secret to a file on the shared `/tmp` volume based on the name provided as the `MOUNT_NAME`
- Container starts up and reads the file at `/tmp/<mount name>` and uses it

In this example, this shows how the init container writes this to tmpfs and then the container reads the value.

Reply:
I would add ENVs to the Busybox container: 
```
      env:
      - name: MOUNT_PATH
        value: "catsndogs"
```
and use the following command: 
```
['sh', '-c', 'echo $(cat /tmp/$MOUNT_PATH) && sleep 3600']
```

Reply:
You should also include error handling to make sure the MOUNT_PATH doesn't include spaces or other invalid characters.